libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,519 @@
+      } else {
+        if (
+          _.get(component, 'vulnerabilities.vulnerability.cve_id') === currcve
+        ) {
+          hascve = true;
+          location += ` ${_.get(component, FILE_PATH_VALUE)}`;
+          _.set(vuln, `paths`, location);
+        }
+      }
+      if (hascve === true) {
+        const proxy = _.cloneDeep(component);
+        components.push(_.omit(proxy, `vulnerabilities`));
+      }
+    }
+    _.set(vuln, `components`, components);
+  })
+  return vulnarr as unknown;
+}
+
+function controlMappingCve(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'cve_id'},
+    title: {path: 'cve_id'},
+    desc: {path: 'cve_summary'},
+    impact: {path: 'impact'},
+    refs: [],
+    tags: {
+      cwe: {path: 'cwe_id'},
+      nist: {
+        path: 'cwe_id',
+        transformer: (value: string) => {
+          if (isArray(value)) {
+            return CWE_NIST_MAPPING.nistFilter(
+              value as string[],
+              DEFAULT_NIST_TAG
+            );
+          } else {
+            return CWE_NIST_MAPPING.nistFilter([value], DEFAULT_NIST_TAG);
+          }
+        }
+      }
+    },
+    source_location: {
+      ref: {path: 'paths'}
+    },
+
+    results: [
+      {
+        path: 'components',
+        status: ExecJSON.ControlResultStatus.Failed,
+        ///code vs message, make sure code_desc is a description
+        code_desc: {transformer: formatSCACodeDesc},
+        start_time: {path: '$.detailedreport.first_build_submitted_date'}
+      }
+    ]
+  };
+}
+
+function controlMappingCwe(
+  severity: number
+): MappedTransform<ExecJSON.Control & ILookupPath, ILookupPath> {
+  return {
+    id: {path: 'categoryid'},
+    title: {path: 'categoryname'},
+    desc: {transformer: formatDesc},
+    descriptions: [
+      {
+        data: {transformer: formatRecommendations},
+        label: 'fix'
+      }
+    ],
+    impact: impactMapping(severity),
+    refs: [],
+    tags: {
+      cweid: {transformer: formatCweData},
+      cweDescription: {transformer: formatCweDesc},
+      nist: {transformer: nistTag}
+    },
+    source_location: {
+      ref: {
+        path: 'cwe',
+        transformer: formatSourceLocation
+      }
+    },
+    results: [
+      {
+        path: 'cwe',
+        pathTransform: getFlaws,
+        status: ExecJSON.ControlResultStatus.Failed,
+        code_desc: {transformer: formatCodeDesc},
+        start_time: {path: '$.detailedreport.first_build_submitted_date'},
+        message: {
+          path: 'exploitability_adjustments.exploitability_adjustment.note'
+        }
+      }
+    ]
+  };
+}
+
+export class VeracodeMapper extends BaseConverter {
+  originalData: unknown;
+  defaultMapping(
+    withRaw = false
+  ): MappedTransform<ExecJSON.Execution & {passthrough: unknown}, ILookupPath> {
+    return {
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion,
+        target_id: ''
```

Comment:
please remove unless we can find a good mapping in the original data, which i feel like we probably could if we conjoined a few fields like detailedreport.modules[0...x].name

Reply:
removed platform